### PR TITLE
fix Issue 24303 - anonymous struct problems when typedef'd in separat…

### PR DESCRIPTION
**Pull request by WalterBright**
…e C files

### CODE REVIEW

File: compiler/test/compilable/test24280.i [hunk 1]
```diff
@@ -4,3 +4,17 @@ struct timespec
...
+
+struct Lang
+{
+    Slice *slices;
+};
+
+void langmap(struct Lang *self)
+{
+    Slice slice = *self->slices;
+}
```
Comment (dkorpel) [inline compiler/test/compilable/test24280.i:20]: This part wasn't in the bugzilla test case, but I assume it still reproduces the issue.

**Review by dkorpel — APPROVED** [review]

**Comment by dkorpel** [comment]
I tested this PR on the C code that found this issue, and it works now, except for a small related thing that I found because I was using both an old and new version of a header file: https://issues.dlang.org/show_bug.cgi?id=24306

**Comment by WalterBright** [comment]
This error:
```
D:\a\1\s\generated\windows\release\32\dmd.exe -conf= -m32omf -Icompilable   -odD:\a\1\s\compiler\test\test_results\compilable -ofD:\a\1\s\compiler\test\test_results\compilable\testdefines_0.obj  -c compilable\testdefines.d 
compilable\testdefines.d(7): Error: undefined identifier `I64`, did you mean variable `F64`?
compilable\testdefines.d(7):        while evaluating: `static assert(I64 == 5)`
```
Appears in other PRs, so it is not related to this PR. @dkorpel could you please have a look and see which PR is the culprit?

**Comment by dkorpel** [comment]
Looks like Rainer found the cause! https://github.com/dlang/dmd/pull/15965

**Pull request merged**